descartes_light/core/include/descartes_light/core/edge_evaluator.h

```diff
@@ -44,7 +49,8 @@ class EdgeEvaluator
    * @return A pair <True/False, Cost>, True if edge is valid, false otherwise. Cost to move from the first vertex to
    * the next
    */
-  virtual std::pair<bool, FloatType> evaluate(const State<FloatType>& start, const State<FloatType>& end) const = 0;
+  virtual std::pair<bool, FloatType> evaluate(const Eigen::Ref<const State<FloatType>>& start,
+                                              const Eigen::Ref<const State<FloatType>>& end) const = 0;
```

Comment:
What's the purpose of passing an `Eigen::Ref` by reference and passing an `Eigen` object by reference? The change seems to be a lateral move that makes the function signature longer without adding any value

Reply:
If you use the Eigen type by reference then the type passed must exactly be that type and if it is not then it will be copied. If you used Eigen::Ref then you will avoid copying where you pass a bloc or other Eigen ref types. For example, if you try to pass data into the function using block, head, tail, row, col, ext.

Reply:
Okay that makes sense. I'm on board